extensions/git/src/git.ts

```diff
@@ -1993,6 +1998,24 @@ export class Repository {
 			child.on('error', cpErrorHandler(e));
 			child.on('exit', onExit);
 		});
+
+		if (opts?.cancellationToken) {
+			const cancellationPromise = new Promise<{ status: IFileStatus[]; statusLength: number; didHitLimit: boolean }>((_, e) => {
+				onceEvent(opts.cancellationToken!.onCancellationRequested)(() => {
```

Comment:
We should dispose this listener if child ends up dying before the event gets fired.

Reply:
Done.